src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java

```diff
@@ -849,13 +858,189 @@ public void given_event_requestHearingRequirementsFeature_when_initiated_verify_
+                ),
+                holdUntilTimeStamp
+            )
+        );
+
+        putEventToRestEndpoint(messageId, s2sToken, updateRequest, true)
+            .then()
+            .statusCode(HttpStatus.CREATED.value())
+            .assertThat()
+            .body("MessageId", equalTo(messageId))
+            .body("Sequence", equalTo(sequence))
+            .body("CaseId", equalTo(caseIdForTask))
+            .body("EventTimestamp", equalTo(updatedEventTimestamp.toString())) // updated
+            .body("FromDlq", equalTo(true)) // updated
+            .body("State", equalTo(MessageState.NEW.name()))
+            .body("MessageContent", equalTo(asJsonString(updateRequest))) // updated
+            .body("Received", notNullValue())
+            .body("DeliveryCount", equalTo(0))
+            .body("HoldUntil", equalTo(holdUntilTimeStamp.toString()))
+            .body("RetryCount", equalTo(0))
+
+            .rootPath("MessageProperties")
+            .body("messageProperty1", equalTo("value1"))
+            .body("messageProperty2", equalTo("value2"));
+    }
+
+    @Test
+    public void should_get_ccd_event_using_test_rest_endpoints() {
+        String messageId = randomMessageId();
+        String caseIdForTask = UUID.randomUUID().toString();
+        String eventInstanceId = UUID.randomUUID().toString();
+
+        EventInformation eventInformation = buildEventInformation(eventInstanceId, caseIdForTask);
+        EventInformationRequest createRequest = new EventInformationRequest(
+            eventInformation,
+            new EventInformationMetadata(
+                Map.of(
+                    "messageProperty1", "value1",
+                    "messageProperty2", "value2"
+                ),
+                holdUntilTimeStamp
+            )
+        );
+
+        Integer sequence = postEventToRestEndpoint(messageId, s2sToken, createRequest)
+            .then()
+            .statusCode(HttpStatus.CREATED.value())
+            .extract()
+            .path("Sequence");
+
+        getEventToRestEndpoint(messageId, s2sToken)
+            .then()
+            .statusCode(HttpStatus.OK.value())
+            .assertThat()
+            .body("MessageId", equalTo(messageId))
+            .body("Sequence", equalTo(sequence))
+            .body("CaseId", equalTo(caseIdForTask))
+            .body("EventTimestamp", equalTo(eventTimeStamp.toString()))
+            .body("FromDlq", equalTo(false))
+            .body("State", equalTo(MessageState.NEW.name()))
+            .body("MessageContent", equalTo(asJsonString(createRequest)))
+            .body("Received", notNullValue())
+            .body("DeliveryCount", equalTo(0))
+            .body("HoldUntil", equalTo(holdUntilTimeStamp.toString()))
+            .body("RetryCount", equalTo(0))
+
+            .rootPath("MessageProperties")
+            .body("messageProperty1", equalTo("value1"))
+            .body("messageProperty2", equalTo("value2"));
+    }
+
```

Comment:
Test to show when mandatory field is missing, we save a message with state unprocessable?